src/System.Text.Json/src/System/Text/Json/JsonReaderState.cs

```diff
@@ -79,9 +85,11 @@ public JsonReaderState(int maxDepth = StackFreeMaxDepth, JsonReaderOptions optio
             _previousTokenType = default;
             _readerOptions = options;
 
-            // Only allocate the stack if the user reads a JSON payload beyond the depth that the _stackFreeContainer can handle.
+            // Only allocate if the user reads a JSON payload beyond the depth that the _allocationFreeContainer can handle.
             // This way we avoid allocations in the common, default cases, and allocate lazily.
-            _stack = null;
+            _bitArray = default;
```

Comment:
Though this also makes JsonState larger, because CustomUncheckedBitArray is larger the reference that's stored today.  And that means any heap object storing JsonState (e.g. an async state machine) is now larger as a result, even if the stack is never used.